From e282d1b08777ea8ba8bc8e411fe266129869ce62 Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Fri, 13 Jun 2025 12:06:48 +0200
Subject: [PATCH 40/55] fwtool: check for EST support if the cert was issued by
 insta

Adds EST certificate validation during firmware upgrade to ensure signed images are only accepted on devices with valid certificates.

Signed-off-by: John Crispin <john@phrozen.org>
---
 include/image-commands.mk                      | 3 ++-
 package/base-files/files/lib/upgrade/fwtool.sh | 4 ++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/include/image-commands.mk b/include/image-commands.mk
index 21d53877f1..30cfd00f97 100644
--- a/include/image-commands.mk
+++ b/include/image-commands.mk
@@ -85,7 +85,8 @@ metadata_json = \
 			"revision": "$(call json_quote,$(REVISION))", \
 			"target": "$(call json_quote,$(TARGETID))", \
 			"board": "$(call json_quote,$(if $(BOARD_NAME),$(BOARD_NAME),$(DEVICE_NAME)))" \
-		} \
+		}, \
+		"est_supported": 1 \
 	}'
 
 define Build/append-metadata
diff --git a/package/base-files/files/lib/upgrade/fwtool.sh b/package/base-files/files/lib/upgrade/fwtool.sh
index 8bd00a3332..eb0465d4de 100644
--- a/package/base-files/files/lib/upgrade/fwtool.sh
+++ b/package/base-files/files/lib/upgrade/fwtool.sh
@@ -51,6 +51,10 @@ fwtool_check_image() {
 	json_get_var compatmessage compat_message
 	[ -n "$imagecompat" ] || imagecompat="1.0"
 
+	if ! est_client fwtool; then
+		return 1
+	fi
+
 	# select correct supported list based on compat_version
 	# (using this ensures that compatibility check works for devices
 	#  not knowing about compat-version)
-- 
2.34.1

